src/reducers/vis-state-updaters.js

```diff
@@ -1524,3 +1530,215 @@ export function updateAnimationDomain(state) {
     }
   };
 }
+
+/**
+ * Update the status of the editor
+ * @memberof visStateUpdaters
+ * @param {Object} state `visState`
+ * @param {string} mode to set to editor to
+ * @return {Object} nextState
+ */
+export const setEditorModeUpdater = (state, {mode}) => ({
+  ...state,
+  editor: {
+    ...state.editor,
+    mode
+  }
+});
+
+/**
+ * Update editor features
+ * @memberof visStateUpdaters
+ * @param {Object} state `visState`
+ * @param {[Object]} features to store
+ * @return {Object} nextState
+ */
+export function setFeaturesUpdater(state, {features = []}) {
+
+  const lastFeature = features.length && features[features.length - 1];
+
+  let newState = {
+    ...state,
+    editor: {
+      ...state.editor,
+      features,
+      mode: lastFeature && lastFeature.properties.isClosed ?
```

Comment:
what is `isClosed`? and why when `isClosed` is true editor mode should be EDIT?

Reply:
isClosed is a property that comes from react-map-gl-draw. It means the feature we are currently drawing is completed, e.g. while drawing a polygon we finally draw all lines.

The abose logic allows the app to detect when the current feature is being completed and switch from POLYGON/RECTANGLE mode to EDIT mode so users can immediately select the latest created feature.

![select-feature](https://user-images.githubusercontent.com/2554552/71479625-731d6b00-27f5-11ea-9fff-bde963bcdf22.gif)